File: one-to-one-videocall/src/main/java/fr/marcworld/webrtcdemo/onetoonevideocall/dtos/UserServerEvent.java
package fr.marcworld.webrtcdemo.onetoonevideocall.dtos;

public class UserServerEvent<T> {

    private UserServerEventCode code;
    private T payload;

    public UserServerEvent() {
    }

    public UserServerEvent(UserServerEventCode code) {
        this(code, null);
    }

    public UserServerEvent(UserServerEventCode code, T payload) {
        this.code = code;
        this.payload = payload;
    }

    public UserServerEventCode getCode() {
        return code;
    }

    public void setCode(UserServerEventCode code) {
        this.code = code;
    }

    public T getPayload() {
        return payload;
    }

    public void setPayload(T payload) {
        this.payload = payload;
    }
}

File: src/main/resources/static/scripts/model/UserServerEvent.js
import UserServerEventCode from './UserServerEventCode.js';

class UserServerEvent {

    /**
     * @param {{code: UserServerEventCode, payload: Object}} properties
     */
    static fromProperties(properties) {
        return new UserServerEvent(properties.code, properties.payload);
    }

    /**
     * @param {UserServerEventCode} code
     * @param {Object} payload
     */
    constructor(code, payload) {
        this.code = code;
        this.payload = payload;
    }
}

export default UserServerEvent;